File: src/main/java/com/oz/office_tastezip/global/exception/GlobalException.java
package com.oz.office_tastezip.global.exception;

import com.oz.office_tastezip.global.response.ResponseCode;
import lombok.Getter;

@Getter
public class GlobalException extends RuntimeException {
    private ResponseCode responseCode;

    public GlobalException() {
    }

    public GlobalException(ResponseCode responseCode) {
        super(responseCode.getMessage());
        this.responseCode = responseCode;
    }

    public GlobalException(ResponseCode responseCode, String message) {
        super(message);
        this.responseCode = responseCode;
    }

    public GlobalException(ResponseCode responseCode, String message, Throwable throwable) {
        super(message, throwable);
        this.responseCode = responseCode;
    }
}

File: src/main/kotlin/com/oz/office_tastezip/global/exception/GlobalException.kt
package com.oz.office_tastezip.global.exception

import com.oz.office_tastezip.global.response.ResponseCode

open class GlobalException : RuntimeException {
    val responseCode: ResponseCode

    constructor(responseCode: ResponseCode) : super(responseCode.message) {
        this.responseCode = responseCode
    }

    constructor(responseCode: ResponseCode, message: String) : super(message) {
        this.responseCode = responseCode
    }

    constructor(responseCode: ResponseCode, message: String, throwable: Throwable) : super(message, throwable) {
        this.responseCode = responseCode
    }
}
